ui/app/src/components/FormsEngine/controls/NodeSelector.tsx

```
@@ -831,47 +846,36 @@ function CreateDataSourcePicker(props: {
 	useEffect(() => {
 		refs.current.onChange?.(null, value);
 	}, [refs, value]);
+
+	const [filteredTypes, setFilteredTypes] = useState(undefined);
+	const filteredTypesRef = useRef(undefined);
+	filteredTypesRef.current = filteredTypes;
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/ui/app/src/components/FormsEngine/controls/NodeSelector.tsx:852:2
  850 | 	const [filteredTypes, setFilteredTypes] = useState(undefined);
  851 | 	const filteredTypesRef = useRef(undefined);
> 852 | 	filteredTypesRef.current = filteredTypes;
      | 	^^^^^^^^^^^^^^^^^^^^^^^^ Cannot update ref during render
  853 | 	const [keywords, setKeywords] = useState<string>('');
  854 | 	const onKeyword$ = useDebouncedInput((keywords) => {
  855 | 		if (!allowedTypes) return;

(react-hooks/refs)